### PR TITLE
feat: parallel tab execution with TabExecutor

### PR DESCRIPTION
**Pull request by YashJadhav21**
## Summary

Adds a `TabExecutor` that enables safe concurrent execution across browser tabs while keeping sequential execution within each tab (required by chromedp). Built on top of #159 (bridge foundation) and #165 (strategy framework).

### Core Engine — `internal/bridge/tab_executor.go` (192 lines)

- Two-phase locking: global semaphore (`chan struct{}`) + per-tab `sync.Mutex`
- Panic recovery per task — one tab's failure doesn't crash the process
- Context cancellation with cleanup goroutines for mutex release
- Auto-scaling: `DefaultMaxParallel() = min(NumCPU*2, 8)`

### Integration — wired into existing Bridge/TabManager layer

- `TabManager.Execute()` delegates to `TabExecutor` (nil-safe fallback)
- `BridgeAPI.Execute()` interface method for handler access
- `Bridge.Execute()` delegation method
- Configurable via `PINCHTAB_MAX_PARALLEL_TABS` env var

### Tests — `internal/bridge/tab_executor_test.go` (1025 lines, 41 tests)

- Unit: execution, error propagation, panic recovery, context cancellation
- Edge cases: empty tabID, nil task, maxParallel=1, negative maxParallel
- Stress: 50 concurrent tasks, rapid create/remove, same-tab contention
- Timing: sequential vs parallel comparison (~4x speedup confirmed)
- All pass with `go test -race` (zero data races)

### Documentation — `docs/parallel-tab-execution.md`

- Architecture diagrams, execution flow, components
- Performance benchmarks and concurrency safety proofs
- Architectural inspiration from Vercel Agent Browser and PR #145

## How It Complements #159 and #165

| Layer | Controls | Mechanism |
|-------|----------|-----------|
| Tab eviction (#159) | Total tab count | `close_oldest` / `close_lru` / `reject` |
| **TabExecutor (this PR)** | **Concurrent execution count** | **Global semaphore + per-tab mutex** |
| Strategy framework (#165) | Request routing | Allocation policies (FCFS, round-robin, random) |

## Files Changed (12 files, +2443 lines)

| File | Change |
|------|--------|
| `internal/bridge/tab_executor.go` | New — core execution engine |
| `internal/bridge/tab_executor_test.go` | New — 41 tests |
| `docs/parallel-tab-execution.md` | New — documentation |
| `internal/bridge/tab_manager.go` | `Execute()`, `Executor()`, `CleanStaleTabs` fix |
| `internal/bridge/api.go` | `Execute()` in BridgeAPI interface |
| `internal/bridge/bridge.go` | `Execute()` delegation method |
| `internal/config/config.go` | `MaxParallelTabs` field |
| `internal/handlers/*_test.go` | Mock updates for `Execute()` |
| `.gitignore` | Exclude test output files |

## Validation

```bash
✅ go build ./...                              # compiles clean
✅ go test -race -count=1 ./internal/bridge/   # all tests pass, zero races

### CODE REVIEW

File: internal/bridge/tab_executor_test.go [hunk 1]
```diff
@@ -0,0 +1,1025 @@
...
+	"errors"
+	"fmt"
+	"runtime"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+)
+
+// --- Unit Tests ---
```
Comment (luigi-agosti) [inline internal/bridge/tab_executor_test.go]: See // --- Unit Tests ---  are clear demarcation of doing more than one thing per file... I think if maybe be great to split the file in 3 ... also small files are agent friendly :)

**Review by luigi-agosti — APPROVED** [review]
